juniper_official/Chassis/report-gres-readiness.rule

```diff
@@ -0,0 +1,141 @@
+/*
+ * Check if slave RE is ready for failover
+ *
+ * In dual routing engine chassis, if graceful-switchover is configured, the
+ * slave routing engine syncs the FIB states from Master RE and declares itself 
+ * ready for failover event.
+ *
+ * If for some reason, Slave does not become ready for failover for significant
+ * amount of time, this rule will report it.
+ */
+iceberg {
+    topic chassis.gres {
+        rule report-gres-readiness {
+            synopsis "Check if slave RE is ready for failover";
+            description "Check whether Failover/GRES is configured and if slave RE is ready for a switchover";
+            sensor chassis-gres-sensor {
+                open-config {
+                    sensor-name /junos/chassis/gres;
+                    frequency 10m;
+                }
+            }
+            field configured-state {
+                sensor chassis-gres-sensor {
+                    path /junos/chassis/gres/configured-state;
+                }
+                type string;
+                description "Failover/GRES configured state";
+            }
+            field ore-present {
+                reference {
+                    path "/topic[topic-name='chassis.gres']/rule[rule-name=check-failover-configured]/ore-present";
+                }
+                type integer;
+                description "Dual RE or not";
+            }
+            field slave-connect-time {
+                sensor chassis-gres-sensor {
+                    path /junos/chassis/gres/slave-connect-time;
+                }
+                type integer;
+                description "Slave connect time";
+            }
+            field slave-kernel-ready {
+                sensor chassis-gres-sensor {
+                    path /junos/chassis/gres/slave-kernel-ready;
+                }
+                type string;
+                description "Whether slave kernel is gres ready or not";
+            }
+            trigger report-gres-readiness {
```

Comment:
1) Provide trigger description and synopsis detailing sensor info.
2) Trigger frequency is must as default frequency is 10s and trigger gets evaluated once in 10s. but here sensor frequecy is 10 mins hence trigger frequency can be greaterthan equal to 10 mins.
e.g. synopsis "Routing engine GRES kpi";
       description "sets health based on Routing engine CPU utilization for 15 minutes";

Reply:
Addressed